src/execution_engine.rs

```diff
@@ -244,12 +305,96 @@ impl ExecutionEngine {
 // want owned modules to drop.
 impl Drop for ExecutionEngine {
     fn drop(&mut self) {
-        forget(self.target_data.take().expect("TargetData should always exist until Drop"));
+        forget(
+            self.target_data
+                .take()
+                .expect("TargetData should always exist until Drop"),
+        );
+    }
+}
 
-        if Rc::strong_count(&self.execution_engine) == 1 {
+impl Clone for ExecutionEngine {
```

Comment:
Should the EE really get a publicly accessible clone method? Clone generally means reallocating a new copy, but here we're just making a new reference to the same EE. Feels a bit counter intuitive to me

Reply:
We end up using the equivalent of `Clone` in a couple places around the code base to create a new reference to the same underlying LLV execution engine so all I did was extract this into an impl for the `Clone` trait. 

I think people will understand this better if we mention an `ExecutionEngine` is actually just a reference-counted pointer to an underlying LLVM execution engine. But I don't mind deleting (or hiding with `#[doc(hidden)]`) this `Clone` impl, it's up to you really.

Reply:
Feel free to add documentation for it; preferably on the clone implementation itself (assuming rustdoc will make this visible, otherwise on the EE itself I guess)

Reply:
Rustdoc lets you override the docs for trait implementations, but I'll probably mention that an `ExecutionEngine` is a reference-counted pointer anyway.